opt4j-core/src/main/java/org/opt4j/core/genotype/DoubleGenotype.java

```diff
@@ -134,6 +134,13 @@ public double getUpperBound(int index) {
 	 *            the number of elements in the resulting genotype
 	 */
 	public void init(Random random, int n) {
+		try {
+			getLowerBound(n - 1);
+			getUpperBound(n - 1);
+		} catch (IndexOutOfBoundsException outOfBoundException) {
+			throw new IllegalArgumentException(
+					"Can not initialize a genotype with " + n + " entries with the specified bounds");
```

Comment:
Could you please rethrow outOfBoundException?
`throw new IllegalArgumentException(
					"Cannot initialize a genotype with " + n + " entries with the specified bounds", outOfBoundException); `